File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n) {
    vector<int> v;
    
    v.push_back(n);
    while(n != 1) {
        if(n % 2 == 0) {
            n /= 2;
            v.push_back(n);
        } else {
            n = 3 * n + 1;
            v.push_back(n);
        }
    }
    
    return v;
}

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.java
import java.util.*;

class Solution {
    public List solution(int n) {
        List<Integer> list = new ArrayList<>();
        
        while(true) {
            if(n == 1) {
                list.add(n);
                break;
            }
            
            if(n % 2 == 0) {
                list.add(n);
                n /= 2;
            }
            else if(n % 2 != 0) {
                list.add(n);
                n = 3 * n + 1;
            };
        }
        
        return list;
    }
}